service/src/main/java/greencity/service/HabitAssignServiceImpl.java

```diff
@@ -868,14 +869,22 @@ public HabitAssignDto cancelHabitAssign(Long habitId, Long userId) {
     }
 
     /**
+     * Method to set {@link HabitAssign} status from inprogress to cancelled.
      * {@inheritDoc}
      */
-    public void deleteHabitAssign(Long habitId, Long userId) {
-        HabitAssign habitAssign = habitAssignRepo.findByUserIdAndHabitId(habitId, userId)
-            .orElseThrow(() -> new NotFoundException(
-                ErrorMessage.HABIT_ASSIGN_NOT_FOUND_WITH_CURRENT_USER_ID_AND_HABIT_ID));
-        userShoppingListItemRepo.deleteByShoppingListItemsByHabitAssignId(habitAssign.getId());
-        habitAssignRepo.delete(habitAssign);
+    public void cancelAllHabitAssign(Long habitId, Long userId) {
+        List<HabitAssign> habitAssigns =
+            habitAssignRepo.findListByUserIdAndHabitIdAndStatusIsInProgress(habitId, userId);
+        if (CollectionUtils.isEmpty(habitAssigns)) {
+            throw new NotFoundException(
+                ErrorMessage.HABIT_ASSIGN_NOT_FOUND_WITH_CURRENT_USER_ID_AND_HABIT_ID_AND_INPROGRESS_STATUS);
```

Comment:
Please add `habitId` and `userId` to the message to specify for which `userId` and `habitId` we could not find `habitAssing`